container/lxd/manager.go

```diff
@@ -287,11 +287,13 @@ func (m *containerManager) networkDevicesFromConfig(netConfig *container.Network
 	if len(netConfig.Interfaces) > 0 {
 		return DevicesFromInterfaceInfo(netConfig.Interfaces, machineID)
 	} else if netConfig.Device != "" {
+		// For the eth0 case do not generate an interface name, instead let
+		// LXD select a host_name for that device.
 		return map[string]device{
 			"eth0": newNICDevice(
 				"eth0",
 				netConfig.Device,
-				makeHostInterfaceName(machineID, 0),
```

Comment:
The commit message is from Achilleas:
```
Assign predictable names to host NICs for LXD container virtual NICs  The NIC naming convention is "machineID-deviceIndex" where machine ID is the juju machine ID (e.g. 0/lxd/1) with the slashes removed. This change not only makes it easier to spot such interfaces in the 'ip a' output and but also will allow juju to enumerate their names when the container is removed (this information is exposed by the LXD API as part of the container's state).
```

Reply:
Yeah, I was speaking to him when I wrote this patch.

Reply:
I wonder if instead of not doing it at all, we did 0lxd1-1-deadbeef

And just tack on a random hex string at the end of the device.

Thoughts?

That still gives us the ability to understand and associate devices with what container they are in, but also means they don't collide.

Then again, we can't *reap* them, because we clearly can have multiple that would map to a similar identity.
But it would bridge the gap between "I have no idea" and "I have an idea but I don't know whether the current container is using it"

Reply:
Note that there is a 15 char limit for NIC names so according to my back of the envelope calculations we can only add a single char (0-9,a-z) suffix; assuming 9999 machines and 256 containers per machine and up to 99 NICs we get the following 14-digit pattern: **XXXXlxdYYYY-ZZ**.

And of course this does not take into account long-lived models where machines come and go and each one gets a monotonically increasing machine ID.

Reply:
Hence why we just decided to drop it and get lxd to just assign it for us.

Reply:
The change looks fine _but_ I think we should investigate further and figure out why `networkDevicesFromConfig` seems (AFAICT from looking at the bug) to be called twice, once with a non-empty NIC list and once with an empty one. 

I am a bit concerned that we will end up in a weird state where we end up with an inconsistent device list result as the first pass would yield a NIC `XXXlxdYYY-0` for `eth0` while the second pass would yield an empty name instead.